pkg/services/provisioning/dashboards/config_reader.go

```diff
@@ -24,10 +24,11 @@ func (cr *configReader) parseConfigs(file os.FileInfo) ([]*DashboardsAsConfig, e
 	}
 
 	apiVersion := &ConfigVersion{ApiVersion: 0}
-	yaml.Unmarshal(yamlFile, &apiVersion)
+	if err := yaml.Unmarshal(yamlFile, &apiVersion); err != nil {
```

Comment:
I think it would be better to just return the error here. There is high chance there will also be some other issue if the version is not right so better to fail early rather than later.

Reply:
Changed logging from warn to error and return the error. I will need to fix broken tests because of this.

Reply:
Looking at the code more closely I see that this was my mistake. Version 0 of the config does not specify it explicitly so it will always error. That is why it was ignored before, not having a version means it was default version 0. Returning the error breaks it for all old configs. I will add a comment there why the handling is as it is.